src/timeline/js/directives/ruler.js

```diff
@@ -158,58 +160,61 @@ App.directive("tlRuler", function ($timeout) {
 
       //watch the scale value so it will be able to draw the ruler after changes,
       //otherwise the canvas is just reset to blank
-      scope.$watch("project.scale + markers.length + project.duration", function (val) {
+      scope.$watch("project.scale + markers.length + project.duration + scrollLeft", function (val) {
         if (val) {
-
           $timeout(function () {
             //get all scope variables we need for the ruler
             var scale = scope.project.scale;
             var tick_pixels = scope.project.tick_pixels;
             var each_tick = tick_pixels / 2;
             // Don't go over the max supported canvas size
-            var pixel_length = Math.min(32767,scope.getTimelineWidth(1024));
 
             //draw the ruler
             var ctx = element[0].getContext("2d");
             //clear the canvas first
             ctx.clearRect(0, 0, element.width(), element.height());
             //set number of ticks based 2 for each pixel_length
-            var num_ticks = pixel_length / 50;
 
             ctx.lineWidth = 1;
             ctx.strokeStyle = "#c8c8c8";
             ctx.lineCap = "round";
 
             //loop em and draw em
-            for (var x = 0; x < num_ticks + 1; x++) {
+            var leftScreenPixels = document.getElementById("scrolling_ruler").scrollLeft;
+            var rulerWidthPixels = document.getElementById("scrolling_ruler").clientWidth;
+            var num_ticks = ~~(rulerWidthPixels / each_tick);
+            var startingTick = ~~(leftScreenPixels / each_tick);
+            for (var x = 0; x < num_ticks + 2; x++) {
               ctx.beginPath();
 
               //if it's even, make the line longer
               var line_top = 0;
-              if (x % 2 === 0) {
+              if ((x + startingTick)% 2 === 0) {
```

Comment:
Nit pick: space before `%` operator

Reply:
Readability is good! I'll take nitpicks.